src/workerd/api/http.c++

```diff
@@ -1169,6 +1198,17 @@ kj::Maybe<kj::String> Request::serializeCfBlobJson(jsg::Lock& js) {
   return cf.serialize(js);
 }
 
+void RequestInitializerDict::validate() {
+  KJ_IF_SOME(c, cache){
+    // Check compatability
+    JSG_REQUIRE(Worker::Api::current().getFeatureFlags().getCacheOptionEnabled(), Error, kj::str(
+      "The 'cache' field on 'RequestInitializerDict' is not implemented."));
+
+    // Check that the string is a supported type
+    auto cacheStruct = getCacheModeFromName(c);
```

Comment:
We ought to be able to enforce the `!= Request::CacheMode::NONE` check here while the `no-cache` and `no-store` support is still pending, correct?

Reply:
In this PR yes. In general changing this seems to cause problems with the returning promise in `fetch` tests.

Reply:
So yes, but then how do we "test" these changes?